mailpoet/lib/API/JSON/v1/Newsletters.php

```diff
@@ -327,6 +328,10 @@ public function showPreview($data = []) {
     $newsletter->setBody(
       json_decode($this->emoji->encodeForUTF8Column($newslettersTableName, 'body', $data['body']), true)
     );
+    // ensure newsletter has hash
+    if (!$newsletter->getHash()) {
+      $newsletter->setHash(Security::generateHash());
+    }
```

Comment:
I see how this fixes the issue with the preview, but I have some concerns about this approach. 
It seems that the source of the problem is that the confirmation email doesn't have the hash in the DB. With this fix the missing hash is added only when a user attempts to get a preview of the email so the sites where a user doesn't click preview will still have the missing hash issue.

Why the hash is not created when the email is added to the DB?
Could the missing hash cause issues in different places in the admin now or in the future (e.g. we add the preview link somewhere else)?
Did you consider an approach in which we ensure that the hash always exists for the confirmation email so that new sites don't suffer from the issue? For the older sites, we could add a migration.